### PR TITLE
i2-season statistics

### LINKED ISSUE #19: Season Statistic - Winningest Coach
Create a method in season statistic for winningest coach.

Name: winningest_coach
Argument: season_id
Description: Name of the Coach with the best win percentage for the season
Return value: string

### LINKED ISSUE #20: Season Statistic - Worst Coash
Create a method in season statistic for worst coach

Name: worst_coach
Argument: season_id
Description: Name of the Coach with the worst win percentage for the season
Return value: string

### LINKED ISSUE #21: Season Statistic - Most Accurate Team
Create a method in season statistic for most accurate team

Name: most_accurate_team
Argument: season_id
Description: Name of the Team with the best ratio of shots to goals for the season
Return value: string

### LINKED ISSUE #22: Season Statistic - Least Accurate Team
Create a method in season statistic for least accurate team

Name: least_accurate_team
Argument: season_id
Description: Name of the Team with the worst ratio of shots to goals for the season
Return value: string

### LINKED ISSUE #23: Season Statistic - Most Tackles
Creates a method in season statistic for most tackles

Name: most_tackles
Argument: season_id
Description: Name of the Team with the most tackles in the season
Return value: string

### LINKED ISSUE #24: Season Statistic - Fewest Tackles
Creates a method in season statistic for fewest tackles

Name: fewest_tackles
Argument: season_id
Description: Name of the Team with the fewest tackles in the season
Return value: string

### PR DESCRIPTION
**Pull request by hashmaster3k**
Creation of Season Statistics module along with the following methods:

- winningest_coach
- worst_coach
- most_accurate_team
- least_accurate_team
- most_tackles
- fewest_tackles

Also added stat tracker test class with passing tests.

### CODE REVIEW

File: lib/season_statistics.rb [hunk 1]
```diff
@@ -78,16 +60,38 @@ def calclulate_tackles_per_team(season_id)
...
+  # Name of the team with the worst ratio of shots to goals for the season
+  def least_accurate_team(season_id)
+    acc_team_id = accuracy_per_team(season_id).min_by {|team, accuracy| accuracy}[0]
+    get_team_name(acc_team_id)
+  end
+
 	# Name of the team with the most tackles in the season
   def most_tackles(season_id)
     tac_team_id = calclulate_tackles_per_team(season_id).max_by {|t_id, tackles| tackles}[0]
-    teams.find {|team| team.team_id.to_i == tac_team_id}.teamname
```
Comment (hashmaster3k) [inline lib/season_statistics.rb]: Moved this and line 90 over to it's own method.

File: lib/season_statistics.rb [hunk 2]
```diff
@@ -6,40 +6,34 @@ def games_from_season(season_id)
     end
   end
 
-  def group_wins_by_coach(season_id)
+  def get_team_name(team_id)
```
Comment (hashmaster3k) [inline lib/season_statistics.rb:9]: Created new method since this code was needed by several methods.

**Comment by ryandbarnett** [comment]
I pulled this branch down locally and tried to run the test harness against it. It wasn't working initially but I was able to put require_relative season statistics in stat tracker and now it's working.

**Comment by hashmaster3k** [comment]
> I pulled this branch down locally and tried to run the test harness against it. It wasn't working initially but I was able to put require_relative season statistics in stat tracker and now it's working.

Strange, I didn't have it in mine and it ran with no issue.

**Comment by ryandbarnett** [comment]
Everything looks great. I approve.

I want to try and implement some mocks and stubs for these tests in the future.

**Review by ryandbarnett — APPROVED** [review]

File: lib/season_statistics.rb [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    games_from_season(season_id).group_by {|game| game.head_coach}
+    # returns {coach: [game1, game2, etc.]} per team
+  end
+
+  def group_by_teams(season_id)
+    games_from_season(season_id).group_by {|game| game.team_id}
+    # returns {team_id: [game1, game2, etc.]} per team
+  end
+
+  def win_percentages_per_coach(season_id)
```
Comment (dhalverson) [inline lib/season_statistics.rb:23]: These look good!

**Review by dhalverson — APPROVED** [review]
Approving, we will need to review the code with the spec harness once we have a majority of the methods written

**Pull request merged**